synapse/storage/end_to_end_keys.py

```diff
@@ -40,7 +40,10 @@ def _set_e2e_device_keys_txn(txn):
                 allow_none=True,
             )
 
-            new_key_json = encode_canonical_json(device_keys)
+            # In py3 we need old_key_json to match new_key_json type. The DB
```

Comment:
s/In py3//.

Reply:
hum I guess actually you're right: it doesn't matter in py2 because `bytes` can be compared to `unicode`